src/Analysis/Engine/Impl/Infrastructure/PriorityProducerConsumer.cs

```diff
@@ -58,7 +58,17 @@ internal sealed class PriorityProducerConsumer<T> : IDisposable {
             _comparer = comparer ?? EqualityComparer<T>.Default;
         }
 
-        public void Dispose() => _disposeToken.TryMarkDisposed();
+        public void Dispose() {
+            if (_disposeToken.TryMarkDisposed()) {
+                lock (_syncObj) {
+                    Debug.Assert(_pendingTasks.Count == 0 || _firstAvailablePriority == _maxPriority);
+                    _pendingTasks.Clear();
+                    foreach (var queue in _queues) {
+                        queue.Clear();
+                    }
+                }
+            };
+        }
 
         public void Produce(T value, int priority = 0) {
             if (priority < 0 || priority >= _maxPriority) throw new ArgumentOutOfRangeException(nameof(priority));
```

Comment:
If you want, you can use `Check` that is now in infra

Reply:
Good to know!